lib/model/message.dart

```diff
@@ -167,33 +168,61 @@ class MessageStoreImpl with MessageStore {
       return;
     }
 
-    if (newTopic == null) {
-      // The `subject` field (aka newTopic) is documented to be present on moves.
-      assert(debugLog('Malformed UpdateMessageEvent: move but no newTopic')); // TODO(log)
+    if (newTopic == null && newStreamId == null) {
+      // At least one of `subject` (aka newTopic) or `new_stream_id` (aka
+      // newStreamId) is present documented to be present on moves.
+      assert(debugLog('Malformed UpdateMessageEvent: move but no newTopic or newStreamId')); // TODO(log)
```

Comment:
```
message: Handle moved messages from UpdateMessageEvent
```

Let's put this change in a separate commit before this one. It looks like it fixes a small but observable bug: in the case where you move a message to a different stream but keep the topic the same, we'll now update `message.editState` as appropriate for a move. 🙂